client/src/elm/Pages/MessagingCenter/View.elm

```diff
@@ -66,46 +66,64 @@ view language currentTime nurseId nurse db model =
+                                |> Maybe.withDefault True
+
+                        runAdoptionSurvey =
+                            diffLastSurveyCurrent ResilienceSurveyAdoption
+
+                        runQuarterlySurvey =
+                            diffLastSurveyCurrent ResilienceSurveyQuarterly
                     in
-                    Dict.get nurseId db.resilienceSurveysByNurse
-                        |> Maybe.andThen RemoteData.toMaybe
-                        |> Maybe.map
-                            (\surveys ->
-                                let
-                                    surveysSorted =
-                                        Dict.values surveys
-                                            |> List.sortWith (sortByDateDesc .dateMeasured)
-
-                                    runSurvey surveyType =
-                                        let
-                                            filterCondition survey =
-                                                -- Run a 3 months survey if 3 month has passed since
-                                                -- last survey was completed.
-                                                Date.diff Months survey.dateMeasured currentDate >= 3
-                                        in
-                                        List.filter (.surveyType >> (==) surveyType) surveysSorted
-                                            |> List.head
-                                            |> Maybe.map filterCondition
-                                            -- We need to run the survey if
-                                            -- it is the first time to use the program.
-                                            |> Maybe.withDefault True
-
-                                    runAdoptionSurvey =
-                                        runSurvey ResilienceSurveyAdoption
-
-                                    runQuarterlySurvey =
-                                        runSurvey ResilienceSurveyQuarterly
-                                in
-                                if runQuarterlySurvey then
-                                    viewQuarterlySurvey language currentDate nurseId model.surveyForm
-
-                                else if runAdoptionSurvey then
-                                    viewAdoptionSurvey language currentDate nurseId model.surveyForm
-
-                                else
-                                    messagingCenterView
-                            )
-                        |> Maybe.withDefault messagingCenterView
+                    if surveyCount == 0 then
+                        if runQuarterlySurvey then
+                            viewQuarterlySurvey language currentDate nurseId model.surveyForm
+
+                        else
+                            viewAdoptionSurvey language currentDate nurseId model.surveyForm
+
+                    else if surveyCount == 3 then
```

Comment:
You're counting here both ResilienceSurveyAdoption and ResilienceSurveyQuarterly.
This is wrong.